src/Symfony/Component/PropertyInfo/PropertyInfoExtractor.php

```diff
@@ -39,17 +41,29 @@ class PropertyInfoExtractor implements PropertyInfoExtractorInterface
     private $accessExtractors;
 
     /**
+     * @var Cache
+     */
+    private $cache;
+
+    /**
+     * @var array
+     */
+    private $arrayCache = array();
+
+    /**
      * @param PropertyListExtractorInterface[]        $listExtractors
      * @param PropertyTypeExtractorInterface[]        $typeExtractors
      * @param PropertyDescriptionExtractorInterface[] $descriptionExtractors
      * @param PropertyAccessExtractorInterface[]      $accessExtractors
+     * @param Cache|null                              $cache
      */
-    public function __construct(array $listExtractors = array(), array $typeExtractors = array(),  array $descriptionExtractors = array(), array $accessExtractors = array())
+    public function __construct(array $listExtractors = array(), array $typeExtractors = array(),  array $descriptionExtractors = array(), array $accessExtractors = array(), Cache $cache = null)
     {
         $this->listExtractors = $listExtractors;
         $this->typeExtractors = $typeExtractors;
         $this->descriptionExtractors = $descriptionExtractors;
         $this->accessExtractors = $accessExtractors;
+        $this->cache = $cache;
```

Comment:
instead of passing the cache here I would prefer using composition instead which decorate the default PropertyInfoExtractor something like https://gist.github.com/aitboudad/12419c2db6d19320c5fa